ohttp-nostr.md

```diff
@@ -0,0 +1,122 @@
+
+* EVENT messages: Encapsulate as BHTTP `POST`. The body MUST contain the event as a JSON string.
+* REQ messages: Encapsulate as BHTTP `GET`. The query string MUST carry the NIP-1 filter, encoded either as a hexified JSON string or URL-encoded JSON (final encoding is TBD). // TODO decide on the encoding // TODO: decide on the query param. REQ messages MUST NOT include a subscription ID, as this leaks linkable metadata.
+
+In order to preserve per-request unlinkability, clients MUST use a fresh ephemeral HPKE key pair per request. Furthermore, clients MUST NOT re-send the same request to the same relay.
+
+### Endpoints
+
+#### Target endpoints
+
+Clients that receive a `404 Not Found` response SHOULD interpret it as an indication that the relay does not support OHTTP. In this case, clients SHOULD fall back to a standard WebSocket connection if they wish to interact with the relay.
+
+* `POST /.well-known/ohttp-gateway` - Submit an OHTTP-encapsulated event.
+
+The HTTP body MUST be a binary payload as defined in RFC 9458. The request MUST include a `Content-Type` header with the value `message/ohttp-req`. Relays MUST decapsulate the payload using their advertised key configuration to obtain the inner BHTTP request.
+
+HTTP Responses MUST only include the encapsulated BHTTP response and they MUST return `200 OK` regardless of the inner BHTTP response.
+
+#### BHTTP inner requests
+
+After decapsulation, relays parse the inner BHTTP request. Two methods are supported. BHTTP responses MAY include status codes other than `200 OK`.
+Relays MUST enforce the request and response size limits advertised in NIP-11. All responses MUST be re-encapsulated and returned to the client using the client's reply key.
+
+##### `POST`
+
+The request body MUST contain a single NIP-1 `EVENT` message.
+
+* If valid, the relay MUST return an empty `202 Accepted` response.
+* If malformed or invalid for any other reason, the relay MUST return `400 Bad Request`.
+* For server-side errors, the relay MUST return `500 Internal Server Error`.
+
+##### `GET`
+
+The request query parameter MUST contain a single NIP-1 `REQ` subscription filter message.
+
+* A filter MUST be included in the query string with the key `filter`. // TODO: decide on the encoding
+* The encoding format for this filter (hexified JSON vs. URL-encoded JSON) remains an open question.
+
+Relays MUST return a `200 OK` response with the inner BHTTP response if the request is valid. The response MUST be formatted as new line delimited JSON.
+
+### Key configuration lifetime
+
+Relays MUST rotate their OHTTP key configurations periodically to limit replay attacks and reduce the impact of key compromise. Key rotation is signaled via the HTTP caching headers (Cache-Control, Expires, ETag) on the response to the key configuration fetch defined in RFC 9540.
```

Comment:
hmm, why MUST and not SHOULD?

my main concern is that key consistency checks get harder with frequent rotations